src/main/java/org/candlepin/subscriptions/tally/roller/BaseSnapshotRoller.java

```diff
@@ -153,68 +165,39 @@ private boolean updateMaxValues(TallySnapshot snap, ProductUsageCalculation calc
         boolean changed = false;
         boolean overrideMaxCheck = Granularity.DAILY.equals(snap.getGranularity());
 
-        changed |= updateMainTotals(overrideMaxCheck, snap, calc);
-        changed |= updatePhysicalTotals(overrideMaxCheck, snap, calc);
-        changed |= updateHypervisorTotals(overrideMaxCheck, snap, calc);
-        return changed;
-    }
-
-    private boolean updateMainTotals(boolean override, TallySnapshot snap, ProductUsageCalculation calc) {
-        boolean changed = false;
-        if (override || mustUpdate(snap.getCores(), calc.getTotalCores())) {
-            snap.setCores(calc.getTotalCores());
-            changed = true;
-        }
-
-        if (override || mustUpdate(snap.getSockets(), calc.getTotalSockets())) {
-            snap.setSockets(calc.getTotalSockets());
-            changed = true;
-        }
-
-        if (override || mustUpdate(snap.getInstanceCount(), calc.getTotalInstanceCount())) {
-            snap.setInstanceCount(calc.getTotalInstanceCount());
-            changed = true;
-        }
+        changed |= updateTotals(overrideMaxCheck, snap, calc, HardwareMeasurementType.TOTAL);
+        changed |= updateTotals(overrideMaxCheck, snap, calc, HardwareMeasurementType.PHYSICAL);
+        changed |= updateTotals(overrideMaxCheck, snap, calc, HardwareMeasurementType.HYPERVISOR);
         return changed;
     }
 
-    private boolean updatePhysicalTotals(boolean override, TallySnapshot snap, ProductUsageCalculation calc) {
+    private boolean updateTotals(boolean override, TallySnapshot snap, ProductUsageCalculation calc,
+        HardwareMeasurementType measurementType) {
```

Comment:
This method is called above with each measurement type, but we are setting the same values for each type which doesn't seem correct to me. The way it is right now, we are setting the calculated overall totals for each measurement type. I think that measurement values that we are setting need to come from the appropriate calculated totals.

TOTAL:
sockets <-- calc.getTotalSockets()
cores <-- calc.getTotalCores()
instances <--  calc.getTotalInstanceCount()

PHYSICAL
sockets <-- calc.getTotalPhysicalSockets()
cores <-- calc.getTotalPhysicalCores()
instances <-- calc.getTotalPhysicalInstanceCount()

HYPERVISOR
sockets <-- calc.getTotalHypervisorSockets()
cores <-- calc.getTotalHypervisorCores()
instances <-- calc.getTotalHypervisorInstanceCount()

Reply:
![facepalm](https://user-images.githubusercontent.com/17975/73554603-09a70280-441a-11ea-8b3a-c83dfe0dfd8e.jpg)

Should be corrected now.  Thanks for catching this.